modules/account-lib/src/coin/sol/ataInitializationTransaction.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { Transaction } from '.';
+import { Entry, TransactionRecipient } from '../baseCoin/iface';
+import { instructionParamsFactory } from './instructionParamsFactory';
+import { InstructionBuilderTypes } from './constants';
+import { getSolTokenFromAddress } from './utils';
+import { AtaInit, Memo, TransactionExplanation } from './iface';
+import BigNumber from 'bignumber.js';
+
+const UNSUPPORTED_TOKEN_NAME = 'UNSUPPORTED_TOKEN';
+const TRANSFER_AMOUNT_UNKNOWN_TEXT = 'TRANSFER_AMOUNT_UNKNOWN';
+
+export class AtaInitializationTransaction extends Transaction {
+  private _tokenAccountRentExemptAmount: string;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  get tokenAccountRentExemptAmount(): string {
+    return this._tokenAccountRentExemptAmount;
+  }
+
+  set tokenAccountRentExemptAmount(amount: string) {
+    this._tokenAccountRentExemptAmount = amount;
+  }
+
+  /** @inheritDoc */
+  loadInputsAndOutputs(): void {
+    if (!this._solTransaction || this._solTransaction.instructions?.length === 0) {
+      return;
+    }
+    const outputs: Entry[] = [];
+    const inputs: Entry[] = [];
+    const instructionParams = instructionParamsFactory(this.type, this._solTransaction.instructions);
+
+    for (const instruction of instructionParams) {
+      if (instruction.type === InstructionBuilderTypes.CreateAssociatedTokenAccount) {
+        const token = getSolTokenFromAddress(instruction.params.mintAddress, this._coinConfig.network);
+        const tokenName = token ? token.name : UNSUPPORTED_TOKEN_NAME;
```

Comment:
ok to leave as is for now, but im wondering if we should throw an error if the token isn't supported.